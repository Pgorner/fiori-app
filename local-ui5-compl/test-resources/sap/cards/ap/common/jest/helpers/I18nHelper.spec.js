/*!
 * SAP UI development toolkit for HTML5 (SAPUI5)
 *  * (c) Copyright 2009-2025 SAP SE. All rights reserved
 */
"use strict";

sap.ui.define(["sap/cards/ap/common/helpers/I18nHelper", "sap/ui/core/UIComponent"], function (sap_cards_ap_common_helpers_I18nHelper, UIComponent) {
  "use strict";

  const resolvei18nTextsForIntegrationCard = sap_cards_ap_common_helpers_I18nHelper["resolvei18nTextsForIntegrationCard"];
  const mI18nMap = {
    APP_TITLE: "Display Settlement Dates",
    APP_SUBTITLE: "Display Settlement Dates",
    HEADER_TITLE: "Header Title",
    HEADER_SUBTITLE: "Header Subtitle",
    UNIT: "USD",
    NUMBER: "500",
    GROUP_TITLE_1: "Group 1",
    GROUP_1_LABEL_1: "Activation Status",
    GROUP_1_VALUE_1: "Condition Control Activation Status",
    GROUP_1_LABEL_2: "Process Category",
    GROUP_1_VALUE_2: "Condition Control Process Category Description",
    GROUP_TITLE_2: "Group 2",
    GROUP_2_LABEL_1: "Sales Organization Description",
    GROUP_2_VALUE_1: "Sales Organization Name",
    GROUP_2_LABEL_2: "Short Description",
    GROUP_2_VALUE_2: "Condition Control Settlement Category Name"
  };
  const sId = "testComponent";
  const oManifest = {
    "sap.app": {
      id: sId,
      type: "application"
    }
  };
  const Component = UIComponent.extend("component", {
    metadata: {
      manifest: oManifest
    },
    createContent() {
      return null;
    }
  });
  const rootComponent = new Component(sId);
  jest.spyOn(rootComponent, "getModel").mockImplementation(yesOrNo => {
    return {
      getText: key => {
        return mI18nMap[key];
      }
    };
  });
  describe("resolvei18nTextsForIntegrationCard", () => {
    test("resolvei18nTextsForIntegrationCard: return updated value of manifest with resolved i18n texts ", () => {
      const cardManifest = {
        _version: "1.15.0",
        "sap.app": {
          id: "objectCard",
          type: "card",
          title: "{{APP_TITLE}}",
          i18n: "",
          subTitle: "{{APP_SUBTITLE}}",
          description: "Description of an Object Card",
          applicationVersion: {
            version: "1.0.0"
          }
        },
        "sap.card": {
          type: "Object",
          extension: "module:sap/cards/ap/common/extensions/BaseIntegrationCardExtension",
          header: {
            data: {
              path: "/content/"
            },
            type: "Numeric",
            title: "{{HEADER_TITLE}}",
            subTitle: "{{HEADER_SUBTITLE}}",
            unitOfMeasurement: "{CndnContrTypeDesc}",
            mainIndicator: {
              unit: "{{UNIT}}",
              number: "{{NUMBER}}"
            }
          },
          content: {
            groups: [{
              title: "{{GROUP_TITLE_1}}",
              items: [{
                label: "{{GROUP_1_LABEL_1}}",
                value: "{{GROUP_1_VALUE_1}}",
                name: "CndnContrActvtnStatusName"
              }, {
                label: "{{GROUP_1_LABEL_2}}",
                value: "{{GROUP_1_VALUE_2}}",
                name: "CndnContrProcessCategoryDesc"
              }]
            }, {
              title: "{{GROUP_TITLE_2}}",
              items: [{
                label: "{{GROUP_2_LABEL_1}}",
                value: "{{GROUP_2_VALUE_1}}",
                name: "SalesOrganizationName"
              }, {
                label: "{{GROUP_2_LABEL_2}}",
                value: "{{GROUP_2_VALUE_2}}",
                name: "CndnContrPrtlSettlmtCatName"
              }]
            }]
          }
        },
        "sap.insights": {
          templateName: "ObjectPage",
          parentAppId: "objectCard",
          cardType: "LEAN_DT",
          versions: {
            ui5: ""
          }
        }
      };
      expect(resolvei18nTextsForIntegrationCard(cardManifest, rootComponent.getModel())).toMatchSnapshot();
    });
    test("resolvei18nTextsForIntegrationCard: return updated value of manifest without resolving i18n texts ", () => {
      const cardManifest = {
        _version: "1.15.0",
        "sap.app": {
          id: "objectCard",
          type: "card",
          title: "APP_TITLE",
          subTitle: "APP_SUBTITLE",
          i18n: "",
          description: "Description of an Object Card",
          applicationVersion: {
            version: "1.0.0"
          }
        },
        "sap.card": {
          type: "Object",
          extension: "module:sap/cards/ap/common/extensions/BaseIntegrationCardExtension",
          header: {
            data: {
              path: "/content/"
            },
            type: "Numeric",
            title: "HEADER_TITLE",
            subTitle: "HEADER_SUBTITLE",
            unitOfMeasurement: "{CndnContrTypeDesc}",
            mainIndicator: {
              unit: "UNIT",
              number: "NUMBER"
            }
          },
          content: {
            groups: [{
              title: "{GROUP_TITLE_1}",
              items: [{
                label: "{GROUP_1_LABEL_1}",
                value: "{GROUP_1_VALUE_1}",
                name: "CndnContrActvtnStatusName"
              }, {
                label: "{GROUP_1_LABEL_2}",
                value: "{GROUP_1_VALUE_2}",
                name: "CndnContrProcessCategoryDesc"
              }]
            }, {
              title: "{GROUP_TITLE_2}",
              items: [{
                label: "{GROUP_2_LABEL_1}",
                value: "{GROUP_2_VALUE_1}",
                name: "SalesOrganizationName"
              }, {
                label: "{GROUP_2_LABEL_2}",
                value: "{GROUP_2_VALUE_2}",
                name: "CndnContrPrtlSettlmtCatName"
              }]
            }]
          }
        },
        "sap.insights": {
          templateName: "ObjectPage",
          parentAppId: "objectCard",
          cardType: "LEAN_DT",
          versions: {
            ui5: ""
          }
        }
      };
      expect(resolvei18nTextsForIntegrationCard(cardManifest, rootComponent.getModel())).toMatchSnapshot();
    });
  });
});
//# sourceMappingURL=I18nHelper.spec.js.map