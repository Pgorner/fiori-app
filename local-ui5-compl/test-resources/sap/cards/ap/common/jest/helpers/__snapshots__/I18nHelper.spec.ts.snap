// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvei18nTextsForIntegrationCard resolvei18nTextsForIntegrationCard: return updated value of manifest with resolved i18n texts  1`] = `
{
  "_version": "1.15.0",
  "sap.app": {
    "applicationVersion": {
      "version": "1.0.0",
    },
    "description": "Description of an Object Card",
    "i18n": "",
    "id": "objectCard",
    "subTitle": "Display Settlement Dates",
    "title": "Display Settlement Dates",
    "type": "card",
  },
  "sap.card": {
    "content": {
      "groups": [
        {
          "items": [
            {
              "label": "Activation Status",
              "name": "CndnContrActvtnStatusName",
              "value": "Condition Control Activation Status",
            },
            {
              "label": "Process Category",
              "name": "CndnContrProcessCategoryDesc",
              "value": "Condition Control Process Category Description",
            },
          ],
          "title": "Group 1",
        },
        {
          "items": [
            {
              "label": "Sales Organization Description",
              "name": "SalesOrganizationName",
              "value": "Sales Organization Name",
            },
            {
              "label": "Short Description",
              "name": "CndnContrPrtlSettlmtCatName",
              "value": "Condition Control Settlement Category Name",
            },
          ],
          "title": "Group 2",
        },
      ],
    },
    "extension": "module:sap/cards/ap/common/extensions/BaseIntegrationCardExtension",
    "header": {
      "data": {
        "path": "/content/",
      },
      "mainIndicator": {
        "number": "500",
        "unit": "USD",
      },
      "subTitle": "Header Subtitle",
      "title": "Header Title",
      "type": "Numeric",
      "unitOfMeasurement": "{CndnContrTypeDesc}",
    },
    "type": "Object",
  },
  "sap.insights": {
    "cardType": "LEAN_DT",
    "parentAppId": "objectCard",
    "templateName": "ObjectPage",
    "versions": {
      "ui5": "",
    },
  },
}
`;

exports[`resolvei18nTextsForIntegrationCard resolvei18nTextsForIntegrationCard: return updated value of manifest without resolving i18n texts  1`] = `
{
  "_version": "1.15.0",
  "sap.app": {
    "applicationVersion": {
      "version": "1.0.0",
    },
    "description": "Description of an Object Card",
    "i18n": "",
    "id": "objectCard",
    "subTitle": "APP_SUBTITLE",
    "title": "APP_TITLE",
    "type": "card",
  },
  "sap.card": {
    "content": {
      "groups": [
        {
          "items": [
            {
              "label": "{GROUP_1_LABEL_1}",
              "name": "CndnContrActvtnStatusName",
              "value": "{GROUP_1_VALUE_1}",
            },
            {
              "label": "{GROUP_1_LABEL_2}",
              "name": "CndnContrProcessCategoryDesc",
              "value": "{GROUP_1_VALUE_2}",
            },
          ],
          "title": "{GROUP_TITLE_1}",
        },
        {
          "items": [
            {
              "label": "{GROUP_2_LABEL_1}",
              "name": "SalesOrganizationName",
              "value": "{GROUP_2_VALUE_1}",
            },
            {
              "label": "{GROUP_2_LABEL_2}",
              "name": "CndnContrPrtlSettlmtCatName",
              "value": "{GROUP_2_VALUE_2}",
            },
          ],
          "title": "{GROUP_TITLE_2}",
        },
      ],
    },
    "extension": "module:sap/cards/ap/common/extensions/BaseIntegrationCardExtension",
    "header": {
      "data": {
        "path": "/content/",
      },
      "mainIndicator": {
        "number": "NUMBER",
        "unit": "UNIT",
      },
      "subTitle": "HEADER_SUBTITLE",
      "title": "HEADER_TITLE",
      "type": "Numeric",
      "unitOfMeasurement": "{CndnContrTypeDesc}",
    },
    "type": "Object",
  },
  "sap.insights": {
    "cardType": "LEAN_DT",
    "parentAppId": "objectCard",
    "templateName": "ObjectPage",
    "versions": {
      "ui5": "",
    },
  },
}
`;
